### DISCUSSION TITLE
Synchronize 2 AnimationPlayer

### DISCUSSION THREAD

**Discussion by pixel-boy** [godotengine/godot-proposals#11185] [2D]
In my project I need to synchronize 2 AnimationPlayer (the animations of a tool / weapons synchronized with the animations of the character).

One of the difficulties is that the character can change tools, and the animations of the tool are therefore managed in its own scene.

With the seek method I had considered placing the position of the anim according to a ratio between 0 and 1, but it does not work as I imagined (I can not have the anim paused and just move in it with seek I have the impression, or the keyframes do not trigger maybe).

I would have liked to know if there were solutions for this kind of problem, or discuss what could be done to improve this.

**Comment by theraot** [comment]
You can have one animation player play the other animation players with animation tracks. Yet, this is not designed to have one animation player loaded dynamically.

To take control of the playback, you need to use both `advance` and `seek` methods.

**Reply by pixel-boy** [reply]
The problem in my case is that the weapon/tool ​​objects are separate scenes, and it's not possible to select their AnimationPlayer to play one of their animations in the character's AnimationPlayer.

I tried `seek` without success but maybe `advance `would give better results in this case.

**Reply by theraot** [reply]
I believe you should be able to get the `AnimationPlayer` with Editable Children. However, with that approach I would probably script it. That is, you can access the `Animation` from code, and change the animation track to point to a different `AnimationPlayer` dynamically.

About `advance` and `seek` I want to make emphasis on **both**.

**Reply by pixel-boy** [reply]
I hadn't considered this option, I'll try that, thanks for your help!

**Reply by pixel-boy** [reply]
`seek` has a parameter `update` which allows to apply the keyframes, I set the speedscale of the 2nd AnimationPlayer to 0 to be able to control it exclusively with my variable:

```gdscript
@export_range(0.0,1.0,0.001) var animation_position :=  0.0:
	set(v):
		animation_position = v
		var anim_length = animation_player.current_animation_length
		animation_player.seek(animation_position/anim_length,true)
```
